### PR TITLE
Add analysis abstraction and implement OpenJ9AnalysisImpl

### PR DESCRIPTION
**Pull request by ShishirH**
Signed-off-by: Shishir Halaharvi <shihala1@in.ibm.com>

### CODE REVIEW

**Comment by ShishirH** [comment]
Please hold off reviewing this until #120 is merged

**Comment by ShishirH** [comment]
@dinogun Now that #120 was merged and the changes rebased, this should be ready for a review

File: src/main/java/com/kruize/analysis/AnalysisImpl.java [hunk 1]
```diff
@@ -75,8 +76,16 @@ public void calculateCpuLimit(MetricsImpl instance)
             return;
         }
 
+        if (instance.getName().contains("petclinic")) {
```
Comment (dinogun) [inline src/main/java/com/kruize/analysis/AnalysisImpl.java:79]: yikes ... is this by mistake, why are we checking for petclinic here ?

File: src/main/java/com/kruize/analysis/runtimes/java/OpenJ9AnalysisImpl.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+            }
+
+            double numberOfMetrics = JavaApplicationMetricsImpl.javaApplicationMetricsMap
+                    .get(application).size();
+            double heapRecommendation = maxHeap;
+
+            if (MathUtil.bytesToMB(heapRecommendation) > metrics.getRssRequests())
+            {
+                double MbToBytes = 1000*1000;
+                heapRecommendation = 0.7 * metrics.getRssRequests() * MbToBytes;
```
Comment (dinogun) [inline src/main/java/com/kruize/analysis/runtimes/java/OpenJ9AnalysisImpl.java:50]: Not sure I understand what is 0.7 here ?

File: src/main/java/com/kruize/analysis/runtimes/java/OpenJ9AnalysisImpl.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+            }
+
+            double numberOfMetrics = JavaApplicationMetricsImpl.javaApplicationMetricsMap
+                    .get(application).size();
+            double nonHeapRecommendation = maxNonHeap;
+
+            if (MathUtil.bytesToMB(nonHeapRecommendation) > metrics.getRssRequests())
+            {
+                double MbToBytes = 1000*1000;
+                nonHeapRecommendation = 0.7 * metrics.getRssRequests() * MbToBytes;
```
Comment (dinogun) [inline src/main/java/com/kruize/analysis/runtimes/java/OpenJ9AnalysisImpl.java:83]: Would be good to add some documentation for each of the functions explaining the logic and the rationale.

File: src/main/java/com/kruize/analysis/AnalysisImpl.java (hunk 1 above)
Comment (ShishirH) [reply src/main/java/com/kruize/analysis/AnalysisImpl.java:79]: Oops, yes looks like I left some test code by mistake. Removing it

Comment (ShishirH) [reply src/main/java/com/kruize/analysis/AnalysisImpl.java]: Removed all test code in latest commit

File: src/main/java/com/kruize/analysis/runtimes/java/OpenJ9AnalysisImpl.java (hunk 3 above)
Comment (ShishirH) [reply src/main/java/com/kruize/analysis/runtimes/java/OpenJ9AnalysisImpl.java]: Added documentation

File: src/main/java/com/kruize/analysis/runtimes/java/OpenJ9AnalysisImpl.java (hunk 2 above)
Comment (ShishirH) [reply src/main/java/com/kruize/analysis/runtimes/java/OpenJ9AnalysisImpl.java]: Created a new variable

**Comment by ShishirH** [comment]
@dinogun Couple of questions:

1. What about the algorithm used? I found that using the max heap value recorded was best for throughput config. Should we instead start using the mean? 
2. In the analyseHeapRecommendation function, there is a codeblock that checks if the heap recommendation is greater than the requests as we discussed. If so, it sets it to 70% of the requests. 
Is that logic necessary for the non-heap part as well? Is there a possibility of the non-heap portion going beyond the requests?

**Comment by ShishirH** [comment]
@dinogun Recommended changes made. Only setting the requests is left, which will be done in the last PR to limit changes here to the analysis.

**Pull request merged**